service/src/java/org/apache/hive/service/cli/operation/OperationManager.java

```diff
@@ -106,58 +114,64 @@ public ExecuteStatementOperation newExecuteStatementOperation(HiveSession parent
     return executeStatementOperation;
   }
 
-  public GetTypeInfoOperation newGetTypeInfoOperation(HiveSession parentSession) {
+  public GetTypeInfoOperation newGetTypeInfoOperation(HiveSession parentSession)
+      throws HiveSQLException {
```

Comment:
Should we create a new exception type that more indicative of the state of HS2?

Reply:
My idea is that our thrift apis are good at handling the HiveSQLException now, and finally we can see HiveSQLException throwing at the client side even when we create another exception type, so if we can pass the server error message to the client side, it's ok. How about introducing another `errorCode` for this?